cmd/snap-confine/seccomp-support.c

```diff
@@ -109,12 +135,94 @@ static void validate_bpfpath_is_safe(const char *path)
 	}
 }
 
+static void sc_cleanup_sock_fprog(struct sock_fprog *prog)
+{
+	free(prog->filter);
+	prog->filter = NULL;
+}
+
+static void sc_must_read_filter_from_file(FILE *file, uint32_t len_bytes,
+					  char *what, struct sock_fprog *prog)
+{
+	if (len_bytes == 0) {
+		die("%s filter may only be empty in unrestricted profiles",
+		    what);
+	}
+	prog->len = len_bytes / sizeof(struct sock_filter);
+	prog->filter = (struct sock_filter *)malloc(len_bytes);
+	if (prog->filter == NULL) {
+		die("cannot allocate %u bytes of memory for %s seccomp filter ",
+		    len_bytes, what);
+	}
+	size_t num_read = fread(prog->filter, 1, len_bytes, file);
```

Comment:
Alternatively we could read len `sizeof(struct sock_filter)` records.

Reply:
Unless, I'm missing something it will bring some extra complexity with no benefit. Please correct me if I'm wrong

Reply:
I mean, fread alread reads N*M things, so reading 1*len(bytes) is equivalent to N * sizeof(record). Fread "kind of" protects you against reading a partial record here.

Reply:
Ah, I think I misunderstood you initially. So you mean making one call to fread with `(prog->len * sizeof(struct sock_filter))` instead of `len_bytes`, right? 

I though you suggested to make several calls to fread, reading sizeof(struct sock_filter) on each of them hehe. Then I have no preference here, whatever looks better to you :)

Reply:
Done in 2ae52cbfa2d8635f7365cbe8e21ad31dd4a02f57